OsmAnd/src/net/osmand/plus/osmedit/OsmEditsFragment.java

```diff
@@ -187,4 +268,133 @@ public OsmandApplication getMyApplication() {
 		return (OsmandApplication) getActivity().getApplication();
 	}
 
+	private void showUploadItemsDialog(){
+		dialog = ProgressImplementation.createProgressDialog(
+				getActivity(),
+				getString(R.string.uploading),
+				getString(R.string.local_openstreetmap_uploading),
+				ProgressDialog.STYLE_HORIZONTAL).getDialog();
+		UploadOpenstreetmapPointAsyncTask uploadTask = new UploadOpenstreetmapPointAsyncTask(dialog, this, remotepoi,
+				remotebug, toUpload.length);
+		uploadTask.execute(toUpload);
+
+		dialog.show();
+	}
+
+
+	public class BackupOpenstreetmapPointAsyncTask extends AsyncTask<OsmPoint, OsmPoint, String> {
```

Comment:
This is copy pasted from different place! The code should be reused and kept unique